great_expectations/self_check/util.py

```diff
@@ -1307,6 +1307,9 @@ def candidate_test_is_on_temporary_notimplemented_list_v3_api(context, expectati
     return False
 
 
+test_connections = set()
```

Comment:
This is strictly required. I noticed redshift connections being slow and that we check if we are able to connect many times. This caches our check so we only make the connection once.